File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/ServiceConfiguration.java
package net.nemerosa.ontrack.model.structure;

import com.fasterxml.jackson.databind.JsonNode;
import lombok.Data;

/**
 * Association between a configuration service ID and an actual configuration data.
 */
@Data
public class ServiceConfiguration {

    private final String id;
    private final JsonNode data;

    public static ServiceConfiguration of(JsonNode node) {
        return new ServiceConfiguration(
                node.get("id").asText(),
                node.get("data")
        );
    }
}

File: ontrack-model/src/main/java/net/nemerosa/ontrack/model/structure/ServiceConfiguration.kt
package net.nemerosa.ontrack.model.structure

import com.fasterxml.jackson.databind.JsonNode
import net.nemerosa.ontrack.model.annotations.APIDescription
import net.nemerosa.ontrack.model.docs.SelfDocumented

/**
 * Association between a configuration service ID and an actual configuration data.
 */
@SelfDocumented
data class ServiceConfiguration(
    @APIDescription("ID of the service")
    val id: String,
    @APIDescription("Configuration of the service")
    val data: JsonNode?
) {
    companion object {
        @JvmStatic
        fun of(node: JsonNode): ServiceConfiguration {
            return ServiceConfiguration(
                node.get("id").asText(),
                node.get("data")
            )
        }
    }
}
